File: src/main/java/com/mateusz/spring_boot_todos/repositories/TodoRepository.java
package com.mateusz.spring_boot_todos.repositories;

import org.springframework.data.jpa.repository.JpaRepository;

import com.mateusz.spring_boot_todos.models.Todo;
import java.util.List;


public interface TodoRepository extends JpaRepository<Todo, Integer> {
    List<Todo> findByUser_Username(String username);
}

File: src/main/kotlin/com/mateusz/todos/kotlin_sboot_todos_backend/repositories/TodoRepository.kt
package com.mateusz.todos.kotlin_sboot_todos_backend.repositories

import com.mateusz.todos.kotlin_sboot_todos_backend.models.Todo
import org.springframework.data.jpa.repository.JpaRepository

interface TodoRepository : JpaRepository<Todo?, Int?> {
    @Suppress("FunctionName") // underscore for postgres
    fun findByUser_Username(username: String?): List<Todo?>?
}